includes/batch.inc

```diff
@@ -0,0 +1,199 @@
+ * Batch functionality for the compound solution pack.
+ */
+
+/**
+ * Creates batch object for deleting the children of a compound.
+ *
+ * @param AbstractObject $compound
+ *   The compound we are deleting from.
+ * @param array $children
+ *   An array of children to delete.
+ *
+ * @return array
+ *   The batch array to be passed to batch_set.
+ */
+function islandora_compound_object_delete_children_batch(AbstractObject $compound, array $children) {
+  $batch = array(
+    'finished' => 'islandora_compound_object_batch_finished',
+    'progress_message' => t('Time elapsed: @elapsed <br/>Estimated time remaining @estimate.'),
+    'error_message' => t('An error has occurred.'),
+    'file' => drupal_get_path('module', 'islandora_compound_object') . '/includes/batch.inc',
+  );
+  $message_parameters = array('@compound' => $compound->label);
+  $child_count = count($children);
+  $batch += array(
+    'operations' => array(
+      array(
+        'islandora_compound_object_delete_children_batch_operation',
+        array($compound, $children),
+      ),
+    ),
+    'title' => format_plural($child_count,
+      'Deleting 1 child from @compound ...',
+      'Deleting @count children from @compound ...', $message_parameters),
+    'init_message' => format_plural($child_count,
+      'Preparing to delete 1 child from @compound ...',
+      'Preparing to delete @count children from @compound ...', $message_parameters),
+  );
+  return $batch;
+}
+
+/**
+ * Starts the batch operation.
+ *
+ * @param AbstractObject $compound
+ *   The compound being deleted.
+ * @param array $children
+ *   The children of the compound being deleted.
+ * @param array $context
+ *   The batch context.
+ *
+ * @return mixed
+ *   The child being processed in this invocation.
+ */
+function islandora_compound_object_start_operation(AbstractObject $compound, array $children, array &$context) {
+  if (!isset($context['sandbox']['progress'])) {
+    $context['sandbox']['progress'] = 0;
+    $context['sandbox']['total'] = count($children);
+    $context['results'] = array(
+      'deleted' => 0,
+      'removed' => 0,
+      'parent' => $compound->label,
+    );
+  }
+  return $children[$context['sandbox']['progress']];
+}
+
+/**
+ * Check if an object has other parents.
+ *
+ * Will return true if object has any relationship to anther object other then
+ * its content model object.
+ *
+ * @param AbstractObject $child
+ *   The child object.
+ * @param AbstractObject $compound
+ *   The parent compound.
+ *
+ * @return bool
+ *   TRUE if the child has other parents, FALSE otherwise.
+ */
+function islandora_compound_object_has_other_parents(AbstractObject $child, AbstractObject $compound) {
+  $relationships = $child->relationships->get();
+  foreach ($relationships as $relationship) {
+    $is_cmodel_relationship = $relationship['predicate']['namespace'] == FEDORA_MODEL_URI
+      && $relationship['predicate']['value'] == 'hasModel';
+    if (!$is_cmodel_relationship
```

Comment:
good, good! Reminded me of a cases of removed cmodels and objects that stayed behind. Never orphans really, more like class-less, which is easier to fix